xen/arch/arm/coproc/coproc.h

```diff
@@ -66,30 +67,33 @@ struct coproc_device {
     struct list_head vcoprocs;
 
     /* coproc callback functions */
-    const struct vcoproc_ops *ops;
+    const struct coproc_ops *ops;
 
     /* scheduler instance for this coproc */
     struct vcoproc_scheduler *sched;
 };
 
 /* coproc callback functions */
-struct vcoproc_ops {
+struct coproc_ops {
     /* callback to perform initialization for the vcoproc instance */
-    struct vcoproc_instance *(*vcoproc_init)(struct domain *,
-                                             struct coproc_device *);
+    int (*vcoproc_init)(struct domain *, struct coproc_device *,
```

Comment:
Why do you need struct coproc_device here? It is delivered with struct vcoproc_instance